### ISSUE TITLE
Music: Add YouTube link

### ISSUE THREAD

**Issue by swapagarwal** [swapagarwal/JARVIS-on-Messenger#258]
Need to find a mapping from spotify track id to youtube id, if possible.

**Comment by bmroach** [comment]
Hi @swapagarwal , can I help out here?

**Comment by horrormyth** [comment]
@bmroach Lets do that

**Comment by swapagarwal** [comment]
@bmroach Sure, go ahead! 😄

**Comment by azie-ginanjar** [comment]
has anyone work on this issue?

**Comment by maferland** [comment]
What is the goal exactly here ? Feed a spotify track id and find the matching youtube id ?

**Comment by akash612** [comment]
Hello @swapagarwal , can I take this one ?

**Comment by thilak007** [comment]
Hi @swapagarwal , is this still open?

**Comment by Jakod96** [comment]
After looking around there doesn't seem to be foolproof method to do it as a 1-1 mapping of the ID to the youtube link, but querrying the youtube API for "ARTIST - SONGNAME" gives very good results.

**Comment by Krafty-Coder** [comment]
Is this done?

**Comment by casdidier** [comment]
I am also interested in this topic is it done ?

**Comment by swapagarwal** [comment]
@Krafty-Coder @casdidier This is still open.

**Comment by Beastjdhr** [comment]
Hey! I'd like to work on this. In which file do I have to add this feature?

**Comment by lovlin-thakkar** [comment]
@swapagarwal Is this issue still open? If yes, can you assist me with this? Thanks.

**Comment by jcconnol** [comment]
Since this is not being assigned to anyone and appears to still be open I will work on this too. I think I will take Jakod96's advice and just query Youtube since Spotify and Youtube are not connected in any way.

**Comment by romankarski** [comment]
@swapagarwal 
hi,
i need done some commit in python for my Python classes at Unicersity.
Is it still open ? Do you think is it proper for beginners to do this commit ? Can You give some advice where to start ? I have founded music.py for spotify

**Comment by CiaranFahy** [comment]
I'll give it a go as I could have found a way to do it, but does anybody know how I could find an artist name and song name from the spotify link

**Comment by zsaeed2** [comment]
This is my first open source project. I was wondering what you guys needed done and how I could contribute to it

**Comment by elkku26** [comment]
I'm new to git, but here's my idea. Could we try to get the spotify page with the id, scrape that page for the name of the song, and use the YT API to look up the name of the song on YouTube and get a link? I know this is very clunky but the first thing that came to mind. Thougts?

**Comment by nuk380y** [comment]
@elkku26 
There are some Python libraries that would be good for that, but it might be easier to get access to the Spotify ([spotipy](https://pypi.org/project/spotipy/)) and YT ([YouTube Data API Documentation](https://developers.google.com/youtube/v3/quickstart/python)) APIs as those will be in a more "machine readable" format.  Song name probably won't be quite enough to ensure accurate results as songs are sometimes covered by other artists, or used as background music by various content creators on YT.

With all of that said, it seems that you're on the right track!  If you can pull the artist name, and the song name from the Spotify API, then it's a matter of figuring out of how to translate that into an old-fashion "Google Hacks" style search query.

**TL;DR**:  The search you would want to make would be something similar to `"{{ artist_name }} - Topic" "{{ track_name }}"`.

YT started aggregating music videos and singles under a topic for the artist when YT was experimenting with YTMusic.  Searching in the web interface for `"{{ artist_name }} - Topic"` (the quotations tell the search function to look for that specific string) would provide all of the songs, singles, promotional videos, and music videos that have been identified as the artist's intellectual property.  Running a search for `"{{ track_name }}"` would provide videos that are titles as close to the specific `track_name`.

For example:  If I were to search Spotify for the song used in the U.S. trailer for "Man on Fire" and find "The Mark Has Been Made" by Nine Inch Nails, I would then search YT with the query `"Nine Inch Nails - Topic" "The Mark Has Been Made"`.

**Comment by nuk380y** [comment]
I would like to assist in this issue if at all possible.  I'm very new to python, but hope at the very least to add some insight in to solving this.

**Comment by elkku26** [comment]
Glad to see this project isn't entirely dead. I'll get home in a few hours and I'll try to get a basic prototype working. I'll update when I've made some progress.

**Comment by elkku26** [comment]
I took a look at the API. It seems that it doesn't support unauthorised requests anymore, meaning you need to have the requests connected to a Spotify account. That makes it a bit harder I think but I'll try to get it working.

**Comment by dkgitcode** [comment]
I have some good experience with the Spotify API;  if this is still open, I'm willing to help with this.

**Comment by Rahul-3000** [comment]
hi @swapagarwal  is the issue still open

**Comment by desikan95** [comment]
Is this still open @swapagarwal ?

**Comment by TheFenrisLycaon** [comment]
Is this open ?

**Comment by atharshah1** [comment]
is this open?
if it is I would like to work on it

**Comment by netra2201** [comment]
Is it open? then I would like to work on it

**Comment by AK2k30** [comment]
is this still active??
If it is then i would like to contribute??

**Comment by PranjalAsthana** [comment]
Is this still open? If it is I would like to contribute.

**Comment by Notlock2011** [comment]
dawg someone close this
